### DISCUSSION TITLE
Setting up EMQX using docker container with ALB on ECS with enabling emqx cluster

### DISCUSSION THREAD

**Discussion by satyamtpatel** [emqx/emqx#7036] [Q&A]
### Subject of the support
Describe your issue here.
We are trying to setup emqx with cluster enabled in the ECS with ALB in AWS. Your documentation is using Classic Load Balancer but it is deprecated in AWS and will not be supported in near future. Can you please provide assistance or steps for setting up emqx on ecs using docker container?
Error/Warning printout if any. 

### Your environment

For EMQ X 4.3 or newer, please provide the log archive generated by [node_dump tool](https://github.com/emqx/emqx/blob/master/bin/node_dump) 

Otherwise please provide below info:
- EMQ X version (e.g. `emqx_ctl status`):
- If cluster (e.g. 3 X 4Core16GB):
- Hardware configuration (e.g. `lscpu`):
- OS (e.g. `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Erlang/OTP version (in case you build emqx from source code): 
- Others:

### LOG File

Missing log file can delay the handling of the issue.

**Comment by zmstone** [comment]
Hi @satyamtpatel 
I’m converting this to a discussion

**Comment by zmstone** [comment]
Hi
As far as I know, AWS ALB has no support for MQTT.
Maybe a NLB (network load balancer) is more suitable.
I’ll ask our cloud team to provide an example.

**Reply by satyamtpatel** [reply]
Any update on above request?

**Reply by zmstone** [reply]
The EMQX cloud team is using NLB.
We’ll publish an article about it, but it’s hard to say when at this point

**Reply by satyamtpatel** [reply]
Do you have any article or suggest steps of setting up cluster using Emqx docker images on ECS?

**Reply by bearrito** [reply]
@zmstone Was an article on this ever published or any more guidance?

**Reply by abertoldo** [reply]
I would like to know, too...Any answer here?

**Reply by id** [reply]
Sorry, no. Our recommended way of deploying EMQX in container environment in AWS is to use EMQX Operator and EKS: https://docs.emqx.com/en/emqx-operator/latest/deployment/on-aws-eks.html.

**Reply by hxk1633** [reply]
Why is EKS the reccomended approach as opposed to using ECS with NLB? I was under the impression that EMQX can be deployed to any service that can host containers...

**Reply by id** [reply]
Of course, we just don't have any experience running EMQX on ECS ourselves, and do not see it often being used for deployment across the users and customers.
